core/domain/html_cleaner.py

```diff
@@ -458,13 +460,127 @@ def convert_to_text_angular(html_data):
+    # in some cases adds </br> closing tag and br is reported as parent
+    # of other tags which produces issues in migration.
+    html_data = html_data.replace('<br>', '<br/>')
+
+    # To convert the rich text content within tabs and collapsible components
+    # to valid CKEditor format. If there is no tabs or collapsible component
+    # convert_tag_contents_to_rte_format will make no change to html_data.
+    html_data = convert_tag_contents_to_rte_format(
+        html_data, feconf.RTE_FORMAT_CKEDITOR)
+
+    soup = bs4.BeautifulSoup(html_data.encode('utf-8'), 'html.parser')
+
+    # Replaces b tag with strong tag.
+    for b in soup.findAll('b'):
+        b.name = 'strong'
+
+    # Replaces i tag with em tag.
+    for i in soup.findAll('i'):
+        i.name = 'em'
+
+    # This block wraps p tag in li tag if the parent of p is ol/ul tag. Also,
+    # if the parent of p tag is pre tag, it unwraps the p tag.
+    for p in soup.findAll('p'):
+        if p.parent.name == 'pre':
+            p.unwrap()
+        elif p.parent.name in ['ol', 'ul']:
+            p.wrap(soup.new_tag('li'))
+
+    # Replaces <p><br><p> with <p>&nbsp;</p>.
+    for br in soup.findAll('br'):
+        parent_tag = br.parent
+        if parent_tag.name == 'p' and parent_tag.get_text() == '':
+            br.replaceWith('&nbsp;')
+
+    # In CKEditor format ol/ul cannot be the parent of ol/ul. So this block
+    # wraps each ol/ul into li tag if the parent of ol/ul is ol/ul. Also if
+    # there are consecutive ol/ul tags this block adds the margin:left according
+    # to the number of ol/ul tags and then remove these consecutive ol/ul tags
+    # to maintain the CKEditor format.
+    list_tags = ['ol', 'ul']
+    for tag_name in list_tags:
+        for tag in soup.findAll(tag_name):
+            if tag.parent.name in list_tags:
+                prev_sib = tag.previous_sibling
+                if prev_sib and prev_sib.name == 'li':
+                    if 'style' in tag.attrs:
+                        del tag['style']
+                else:
+                    margin_value = 40
+                    if 'style' in tag.attrs:
+                        style = tag['style']
+                        match = re.search('margin-left:[0-9]+px;', style)
+                        margin_value += int(
+                            style[match.start() + 12: match.end() - 3])
+
+                    for child in tag.children:
+                        child_margin_value = margin_value
+                        if 'style' in child.attrs:
+                            style = child['style']
+                            match = re.search('margin-left:[0-9]+px;', style)
+                            child_margin_value += int(
+                                style[match.start() + 12: match.end() - 3])
+                        child['style'] = 'margin-left:%dpx;' % margin_value
+                    tag.unwrap()
+
+    oppia_block_components = [
```

Comment:
If I recall, you have a similar or same constant defined in the textangular migration function right? Try to share these constants and not repeat them.

Reply:
Yeah, done!